### DISCUSSION TITLE
Migrate to TS 5

### DISCUSSION THREAD

**Discussion by Raiden1411** [wagmi-dev/abitype#124] [General]
Since viem recently migrated to TS 5 and `NodeNext`, I was wondering if it would make sense to do the same here. 

What do you think @tmm?

**Comment by tmm** [comment]
Definitely! Would be great if abitype's build and typechecking set up was the same.

Probably want to wait on using some TypeScript v5 backwards incompatible features (like [`const` type parameters](https://devblogs.microsoft.com/typescript/announcing-typescript-5-0/#const-type-parameters)) until viem and wagmi's minimum versions are bumped to v5 as well.

**Reply by Raiden1411** [reply]
Agreed. For now the focus will just be updating the typechecking and build. The rest will come in due time once everything is ready
